### PR TITLE
Tests for Offer and Contract

### PR DESCRIPTION
**Pull request by cudkowicz**
Adds testing for the post method in the offer file.

### CODE REVIEW

File: esi_leap/tests/api/controllers/v1/test_offer.py [hunk 1]
```diff
@@ -27,6 +27,10 @@ def create_test_offer(context):
     return o
 
 
+def create_test_offer_data():
+    return {1: 'test', 2: 'offer'}
```
Comment (tzumainn) [inline esi_leap/tests/api/controllers/v1/test_offer.py:31]: Ah, we probably want to simulate actual test offer data more closely - something like:
```
{
   "resource_type": "test_node",
   "resource_uuid": "some-uuid",
   "start_time": "some_time_string"
}
```

Comment (cudkowicz) [reply esi_leap/tests/api/controllers/v1/test_offer.py:31]: Ill add this in for offer. Someone made a ton of changes to contract so I figured I would make sure this works and then sift through the changes in the test to see where I should add it and such.

File: esi_leap/tests/api/controllers/v1/test_contract.py [hunk 2]
```diff
@@ -62,6 +63,14 @@
 random_project = 'randomid'
 
 
+def create_test_contract_data():
+    return {
+   "uuid": "test_uuid",
+   "offer_uuid": "some-uuid",
```
Comment (tzumainn) [inline esi_leap/tests/api/controllers/v1/test_contract.py:69]: we should be consistent in how we write our fake uuids "-" or "_"

Comment (cudkowicz) [reply esi_leap/tests/api/controllers/v1/test_contract.py:69]: oops that was a typo

File: esi_leap/tests/api/controllers/v1/test_offer.py [hunk 3]
```diff
@@ -27,6 +27,15 @@ def create_test_offer(context):
     return o
 
 
+def create_test_offer_data():
+
+    return {   
+        "resource_type": "test_node",
+        "resource_uuid": "some-uuid",
```
Comment (tzumainn) [inline esi_leap/tests/api/controllers/v1/test_offer.py:34]: Here too :)

**Comment by cudkowicz** [comment]
This provides tests for the post method in both offer and contract by adding post_json to base.py and using post_json.

**Comment by tzumainn** [comment]
Okay, the code looks good now - however I think it might not quite work. I'm getting both pep8 and py37 errors - you may want to check those out!

**Pull request merged**